modules/tensor_mechanics/src/materials/ADComputeIncrementalShellStrain.C

```diff
@@ -234,11 +240,15 @@ ADComputeIncrementalShellStrain::computeProperties()
 
       (*_total_strain[j])[i] = (*_total_strain_old[j])[i] + (*_strain_increment[j])[i];
 
+      RankTwoTensor contravariant_transformation_matrix =
```

Comment:
Before we do that, I think we should do some performance profiling, both for the covariant matrix and its inverse. Let's leave this the way it is because this corrects the bug, and then let's do some performance studies later.

Reply:
I have already pushed the code with _contravariant_transformation_matrix as a stateful property.